model/common/src/icon4py/model/common/metrics/metrics_factory.py

```diff
@@ -0,0 +1,554 @@
+vwind_impl_wgt_k = constant_field(icon_grid, init_val, dims.CellDim, dims.KDim)
+k_lev = gtx.as_field((dims.KDim,), np.arange(nlev, dtype=gtx.int32))
+e_lev = gtx.as_field((dims.EdgeDim,), np.arange(icon_grid.num_edges, dtype=gtx.int32))
+e_owner_mask = grid_savepoint.e_owner_mask()
+
+fields_factory = factory.FieldsFactory()
+
+fields_factory.register_provider(
+    factory.PrecomputedFieldsProvider(
+        {
+            "height_on_interface_levels": interface_model_height,
+            "cell_to_edge_interpolation_coefficient": c_lin_e,
+            cf_utils.INTERFACE_LEVEL_STANDARD_NAME: k_index,
+            "vct_a": vct_a,
+            "theta_ref_mc": theta_ref_mc,
+            "exner_ref_mc": exner_ref_mc,
+            "wgtfac_c": wgtfac_c,
+            "c_refin_ctrl": c_refin_ctrl,
+            "e_refin_ctrl": e_refin_ctrl,
+            "dual_edge_length": dual_edge_length,
+            "tangent_orientation": tangent_orientation,
+            "inv_primal_edge_length": inv_primal_edge_length,
+            "inv_dual_edge_length": inv_dual_edge_length,
+            "cells_aw_verts_field": cells_aw_verts_field,
+            "vwind_impl_wgt_full": vwind_impl_wgt_full,
+            "vwind_impl_wgt_k": vwind_impl_wgt_k,
+            "k_lev": k_lev,
+            "e_lev": e_lev,
+            "e_owner_mask": e_owner_mask,
+        }
+    )
+)
+
+
+height_provider = factory.ProgramFieldProvider(
+    func=mf.compute_z_mc,
+    domain={
+        dims.CellDim: (
+            horizontal._local(dims.CellDim),
+            horizontal._end(dims.CellDim),
+        ),
+        dims.KDim: (0, nlev),
+    },
+    fields={"z_mc": "height"},
+    deps={"z_ifc": "height_on_interface_levels"},
+)
+fields_factory.register_provider(height_provider)
+
+compute_ddqz_z_half_provider = factory.ProgramFieldProvider(
+    func=mf.compute_ddqz_z_half,
+    deps={
+        "z_ifc": "height_on_interface_levels",
+        "z_mc": "height",
+        "k_index": cf_utils.INTERFACE_LEVEL_STANDARD_NAME,
+    },
+    domain={
+        dims.CellDim: (
+            icon_grid.start_index(cell_domain(horizontal.Zone.LOCAL)),
+            icon_grid.end_index(cell_domain(horizontal.Zone.LOCAL)),
+        ),
+        dims.KDim: (0, nlev + 1),
+    },
+    fields={"ddqz_z_half": "ddqz_z_half"},
+    params={"nlev": nlev},
+)
+fields_factory.register_provider(compute_ddqz_z_half_provider)
+
+ddqz_z_full_and_inverse_provider = factory.ProgramFieldProvider(
+    func=mf.compute_ddqz_z_full_and_inverse,
+    deps={
+        "z_ifc": "height_on_interface_levels",
+    },
+    domain={
+        dims.CellDim: (
+            icon_grid.start_index(cell_domain(horizontal.Zone.LOCAL)),
+            icon_grid.end_index(cell_domain(horizontal.Zone.LOCAL)),
+        ),
+        dims.KDim: (0, nlev),
+    },
+    fields={"ddqz_z_full": "ddqz_z_full", "inv_ddqz_z_full": "inv_ddqz_z_full"},
+)
+fields_factory.register_provider(ddqz_z_full_and_inverse_provider)
+
+divdamp_trans_start = 12500.0
```

Comment:
Where do these parameters come from, I mean the values do you know that?

Reply:
I inherited them from the tests, I found some of them by doing a global search in the fortran code. I agree that they should be re-factored though